### PR TITLE
fix: improve doc item typing

### PR DESCRIPTION
**Pull request by vagenas**

### CODE REVIEW

File: docling_core/types/doc/document.py [hunk 1]
```diff
@@ -1272,9 +1294,25 @@ def export_to_document_tokens(
...
+
+
+ContentItem = Annotated[
+    Union[
+        TextItem,
+        SectionHeaderItem,
+        ListItem,
+        PictureItem,
+        TableItem,
+        KeyValueItem,
```
Comment (PeterStaar-IBM) [inline docling_core/types/doc/document.py:1313]: add FormItem

**Review by PeterStaar-IBM — CHANGES_REQUESTED** [review]
I think you forgot the FormItem in the ContentItem

File: docling_core/types/doc/document.py (hunk 1 above)
Comment (vagenas) [reply docling_core/types/doc/document.py:1313]: Well spotted, added now.

File: docling_core/types/doc/document.py [hunk 2]
```diff
@@ -644,8 +659,10 @@ def export_to_document_tokens(
 class SectionHeaderItem(TextItem):
     """SectionItem."""
 
-    label: typing.Literal[DocItemLabel.SECTION_HEADER] = DocItemLabel.SECTION_HEADER
-    level: LevelNumber
+    label: typing.Literal[DocItemLabel.SECTION_HEADER] = (
+        DocItemLabel.SECTION_HEADER  # type: ignore[assignment]
+    )
+    level: LevelNumber = 1
```
[the remark this answers is not in the document]
Comment (vagenas) [reply docling_core/types/doc/document.py:665]: Yes, `label` is now non-overlapping — and is actually used as the discriminator field in `ContentItem` further below.

**Review by dolfim-ibm — COMMENTED** [review]
Can we easily enforce the type also in the `add_text()` method?

**Comment by vagenas** [comment]
> Can we easily enforce the type also in the `add_text()` method?

Technically it will now be "enforced" when the `TextItem` is created within that method.

If you mean in terms of reflecting it to the typing of the `label` param: if we invest more efforts in that area, I would rather do it in a way that reuses the type validation of the various `DocItems`, e.g. have a single `add_node()`, but I see that as requiring some redesign, and in general outside the scope of this PR.

**Review by PeterStaar-IBM — APPROVED** [review]
Awesome, let's get it merged!

**Review by dolfim-ibm — APPROVED** [review]

**Pull request merged**